[patch] use deprecate from pyiron_snippets

liamhuber opened the pull request.

liamhuber committed:
[patch] use deprecate from pyiron_snippets

jan-janssen commented on pyiron_atomistics/atomistics/job/atomistic.py:
```diff
@@ -17,8 +17,8 @@
     GenericJob as GenericJobCore,
-    deprecate,
 )
+from pyiron_snippets.deprecate import deprecate
```
The import feels a bit redundant `from pyiron_snippets.deprecate import deprecate`. Can we change it to `from pyiron_snippets import deprecate` ?

liamhuber replied:
Certainly not on this version of snippets. Personally, to maintain the idea that snippets is exceedingly lightweight and that all its pieces are independent, I prefer to leave its `__init__` empty. But I'm not totally closed to it so feel free to open an issue on snippets and see what the others think

jan-janssen commented on pyiron_atomistics/atomistics/master/phonopy.py:
```diff
@@ -18,7 +18,8 @@
 )
-from pyiron_base import state, JobGenerator, deprecate
+from pyiron_base import state, JobGenerator
+from pyiron_snippets.deprecate import deprecate
 from pyiron_snippets.import_alarm import ImportAlarm
```
Especially this line would be much cleaner as: 
```suggestion
from pyiron_snippets import deprecate, ImportAlarm
```

liamhuber replied:
As mentioned above, I don't want this because if snippets grows I don't want to burn time importing everything in it just because you use one thing in it. But tbh I'm not sure that even with a large module this burns significant time, so if you have data to prove it won't impact performance down the line, then I'm open to it.

pmrv replied:
The current way keeps the different submodules of snippets and their API uncoupled as is intended and so I think it is strongly preferable to importing/exporting everything in the top module.  Especially since it would be for purely aesthetic reasons.  `from pyiron_snippets.deprecate import *` would work just as well, if it's really unbearable.

liamhuber committed:
[patch] replace logger with snippets version

i.e. pyiron_base.state.logger.logger with pyiron_snippets.logger.logger. The original location leaves a re-direct for backwards compatibility.

jan-janssen committed:
Merge pull request #1438 from pyiron/snippets_logger

[patch] replace logger with snippets version
